collections/nemo_nlp/nemo_nlp/data/datasets/utils.py

```diff
@@ -547,6 +547,134 @@ def merge(data_dir, subdirs, dataset_name, modes=['train', 'test']):
     return outfold, none_slot
 
 
+def get_intent_query_files(path):
+    files = []
+    # r=root, d=directories, f = files
+    for r, d, f in os.walk(path):
+        for file in f:
+            if '_usersays_en.json' in file:
+                files.append(os.path.join(r, file))
+    return files
+
+
+def get_intents_and_slots(files, slot_labels):
+
+    intent_names = []
+    intent_queries = []
+    slot_tags = []
+
+    for index, file in enumerate(files):
+        intent_names.append(os.path.basename(file).split('_usersays')[0])
+
+        with open(file) as json_file:
+            intent_data = json.load(json_file)
+            for query in intent_data:
+                querytext = ""
+                slots = ""
+                for segment in query['data']:
+                    querytext = ''.join([querytext, segment['text']])
+                    if 'alias' in segment:
+                        for word in segment['text'].split():
+                            slots = ' '.join([
+                                    slots,
+                                    slot_labels.get(segment['alias'])])
+                    else:
+                        for word in segment['text'].split():
+                            slots = ' '.join([slots, slot_labels.get('O')])
+                querytext = f'{querytext.strip()}\t{index}\n'
+                intent_queries.append(querytext)
+                slots = f'{slots.strip()}\n'
+                slot_tags.append(slots)
+                # TO DO
+                # WARNING - Slot tag length and word length doesnt
```

Comment:
use logger.warning to display warning. if this pr isn't finished, put WIP in the front

Reply:
done.